pkg/apis/batch/v1/defaults.go

```diff
@@ -38,7 +40,11 @@ func SetDefaults_Job(obj *batchv1.Job) {
 		obj.Spec.Parallelism = utilpointer.Int32(1)
 	}
 	if obj.Spec.BackoffLimit == nil {
-		obj.Spec.BackoffLimit = utilpointer.Int32(6)
+		if obj.Spec.BackoffLimitPerIndex != nil {
```

Comment:
@deads2k and @liggitt pointed out in my review that we should have conditional defaulting for alpha features.

I guarded this under a feature flag in https://github.com/kubernetes/kubernetes/pull/119301/files#diff-850ee9a00a92fc90e918491d581eac3c5b08208c8f6d9f88fa6a8cddee384fda so maybe that is needed?

Reply:
IIUC the difference between `backoffLimitPerIndex` and `podRecreationPolicy` is that, a value for `podRecreationPolicy` is required when the feature is enabled. Whereas, for backoffLimitPerIndex, the feature is opt-in, so even if the feature gate is enabled, leaving `.spec.backoffLimitPerIndex=nil` is supported.

Reply:
So, we don't default `backoffLimitPerIndex`, we default `backoffLimit` to max int32, when a user specified `backoffLimitPerIndex` only.

Reply:
So if this feature gets turned off, we would default to `utilpointer.Int32(math.MaxInt32)` even if the feature was disabled?

Reply:
Since this is not a new field, it's acceptable.